code/items/weapons/Knife.cs

```diff
@@ -61,17 +61,28 @@ public override void AttackSecondary()
 
         public override void AttackPrimary()
         {
-            ShootEffects();
+            (Owner as AnimEntity).SetAnimBool("b_attack", true);
+
+            using (Prediction.Off())
+            {
+                ShootEffects();
```

Comment:
What's the reason behind turning off prediction here?

My concern would be, since we use prediction for the actual shot on the server, and not on client side, the effects may be out of line.

Reply:
Calling ClientRpc's without turned off predictions, crashes can happen. Until that change, I didn't got any crash anymore, but could be random. `ShootEffects` is a ClientRpc